src/hooks/viewport/useMediaQuery.tsx

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useEffect, useState } from "react";
+
+type MediaType = "isMobile" | "isTablet" | "isLaptop";
+const mediaBreakpoints: Record<MediaType, string> = {
+  isMobile: "(max-width: 320px)",
+  isTablet: "(max-width: 768px)",
+  isLaptop: "(max-width: 1024px)",
+};
+export const useMediaQuery = () => {
```

Comment:
This looks interesting. Kind of like it 👍 even though I haven't spent time looking at the details how and what it does 😄 

That being said, I'm still on the fence there isn't a way to offload stuff like this to the browser by more comprehensive CSS later on, so I suggest adding comments explaining **the why**, maybe an example and probably a `@debt` tag somewhere if we think there is an alternative without JS doing the calculation for us.

Reply:
this one is currently used in `NavOverlay` only, it's hard to do that with SCSS since we need to change the display logic for some components ( create some sort of paging for mobile markup ), so this hook was the way to solve it
think we'll apply it somewhere else in the future too